### DISCUSSION TITLE
RuCore / Системы кодирования / Документы удостоверяющие личность МЗРФ

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#241] [Минимальный набор артефактов]
//--------------------------------------
//страница: Руководство RuCore / Системы кодирования / Документы удостоверяющие личность МЗРФ
//--------------------------------------

```
Instance:   core-ns-nsi-identity-document
InstanceOf: NamingSystem
Usage: #definition

* name =   "core-ns-nsi-identity-document"

* status = #active

* kind = #codesystem

* date = "2023-10-23"

* description = " Справочник НСИ Минздрава <Документы, удостоверяющие личность>, может быть известен под следующими идентификаторами: OID: 1.2.643.5.1.13.13.99.2.48, Дополнительный OID: 1.2.643.5.1.13.13.11.1011, 1.2.643.5.1.13.2.1.1.736, предпочтительно URI: http://fhir.ru/core/CodeSystem/core-cs-nsi-identity-document "

* uniqueId[0]
  * type = #oid
  * value = "1.2.643.5.1.13.13.99.2.48"
  * comment = "Основной"

* uniqueId[1]
  * type = #oid
  * value = "1.2.643.5.1.13.13.11.1011"
  * comment = "Дополнительный"

* uniqueId[2]
  * type = #oid
  * value = "1.2.643.5.1.13.2.1.1.736"
  * comment = "Дополнительный"

* uniqueId[3]
  * type = #uri
  * preferred = true
  * value = "http://fhir.ru/core/CodeSystem/core-ns-nsi-identity-document"

* uniqueId[4]
  * type = #uri
  * value = "http://nsi.rosminzdrav.ru/#!/refbook/1.2.643.5.1.13.13.99.2.48"


Alias: $МЗРФ_Справочник_ДокументыУдостоверяющиеЛичность = Core_Cs_Nsi_Identity_Document
CodeSystem: Core_Cs_Nsi_Identity_Document
Id:         core-cs-nsi-identity-document
Title: "Документы удостоверяющие личность"
Description: "НСИ МЗ РФ справочник [документы удостоверяющие личность](http://fhir.ru/core/CodeSystem/core-cs-nsi-identity-document)"

* ^experimental = false

* ^caseSensitive = false

* ^content = #fragment

* #100 "Общегражданские документы"
  * #1 "Паспорт гражданина Российской Федерации"
  * #42 "Заграничный паспорт гражданина Российской Федерации"
  * #2 "Заграничный паспорт гражданина РФ с электронным носителем информации"
  * #5 "Временное удостоверение личности гражданина РФ"
  * #4 "Служебный паспорт гражданина РФ"
  * #6 "Свидетельство о рождении"
  * #24 "Медицинское свидетельство о рождении" "для детей в возрасте до 1 месяца"
  * #43 "Свидетельство о смерти"
* #400 "Специальные документы, связанные с удостоверением личности"
  * #36 "Водительское удостоверение"
```

Методические рекомендации по данному типу артефакта: https://github.com/fhir-ru/core/discussions/249

**Comment by Vrodvi** [comment]
# Замечание 1.

Указанный предпочтительный идентификатор справочника:
```
"http://fhir.ru/core/cs/core-cs-nsi-identity-document"
```
отличается от идентификатора самого справочника, который формируется по правилу составления канонических URL'ов как базовыйUrl+resourceType+id, т.е. в данном случае - это будет:
```
http://fhir.ru/core/CodeSystem/nsi-identity-document
```
(при условии, что базовый url - `http://fhir.ru/core`)

**Reply by eugenekogan-spb** [reply]
Исправлено.

**Reply by Vrodvi** [reply]
Атрибут identifier в ресурсе CodeSystem предназначен для указания "дополнительных" идентификаторов, а указан в нём - основной идентификатор (основной идентификатор - это значение url атрибута с каноническим url (если его явно не указывать, то он формируется автоматически))
`* ^identifier.value = "http://fhir.ru/core/CodeSystem/nsi-identity-document"`

![image](https://github.com/fhir-ru/core/assets/92928137/7e2185b5-d610-428f-a0ab-5b802f11a45f)

Рекомендуется либо убрать атрибут `identifier`, либо указать в нём все остальные "дополнительные" идентификаторы.

Саму ситуацию, когда система кодирования имеет более одного идентификатора, можно описать в документе с методологическими рекомендациями к данному типу артефакта, указав, к примеру, что все дополнительные идентификаторы можно будет найти в NamingSystem и поэтому они не указываются в самом CodeSystem (или альтренативно - укзаать, что основной иднетификатор справочника - это его канонический url, а все остальные должны указываться в атрибуте identifier).

**Reply by eugenekogan-spb** [reply]
Остался не оговоренным случай, когда основной идентификатор не совпадает с каноническим url.

Идентификатор убрал. Завел страничку с методическими рекомендациями - https://github.com/fhir-ru/core/discussions/249.

**Reply by Vrodvi** [reply]
> Остался не оговоренным случай, когда основной идентификатор не совпадает с каноническим url.

В этом случае придётся url для CodeSystem всё таки переопределять на предпочтительный.

Вот характерный пример с таким переопределением от HL7 - для CodeSystem SNOMED url явно переопределён на предпочтительный, который не соответствует каноническому:
https://terminology.hl7.org/CodeSystem-v3-snomed-CT.json.html
https://terminology.hl7.org/NamingSystem-v3-snomed-CT.html

![image](https://github.com/fhir-ru/core/assets/92928137/0997cb2c-9c60-4d5a-8467-6d093591864f)

Тут ещё вот какой загадочный момент - для SNOMED указано два идентификатора как предпочтительные, хотя url CodeSystem'а равен только одному из них, а второй указан в Identifier.

Что всё это означает - пока не совсем понятно, но в очередной раз FHIR подтвердил, что он очень гибкий стандарт!

**Reply by Vrodvi** [reply]
Но, конкретно для нас это пока не критично, т.к. наш URL - это канонический и он же предпочтительный.
Поэтому данное замечание закрываем.

**Reply by Vrodvi** [reply]
Замечание "1." закрыто.

**Comment by Vrodvi** [comment]
@eugenekogan-spb 

Фрагмент с указанием конкретной версии лучше убрать из перечня идентификаторов справочника:
```
* uniqueId[4]
  * type = #uri
  * value = "http://nsi.rosminzdrav.ru/#!/refbook/1.2.643.5.1.13.13.99.2.48/version/7.1"
```

Как раз хорошо, что версия не будет фигурировать в описании артефактов на сайте ру-кора, т.к. за актуальностью версий нам всё равно не уследить, а фиксировать какие-то конкретные версии будет скорее вредно, чем полезно (должен быть весомый повод подобное ограничение вводить, а у нас их нет).

Конкретно для данного справочника версия `7.1` уже не актуальна:
![image](https://github.com/fhir-ru/core/assets/92928137/70b75fcc-dd0d-4d6d-8421-63900cc7426a)
 
А для ссылки на "последнюю" версию справочника в ФНСИ есть специальный вид ссылки (уж лучше на такое заменить):
```
https://nsi.rosminzdrav.ru/dictionaries/1.2.643.5.1.13.13.99.2.48/passport/latest
```

**Reply by Vrodvi** [reply]
не исправлено

**Reply by eugenekogan-spb** [reply]
Исправлено на 
 * value = "http://nsi.rosminzdrav.ru/#!/refbook/1.2.643.5.1.13.13.99.2.48"

**Comment by Vrodvi** [comment]
На автоматически сгенерированном сайте множество проблем с этим справочником, см.:

![image](https://github.com/fhir-ru/core/assets/92928137/abc23856-a2d4-4359-b62a-579268039b4d)


1) набор значений не "разворачивается" (expantion) из-за того что наполнение справочника указано как "пример", а не как "фрагмент" или "полное наполнение".
тип "пример" так же приводит к невозможности для инструментария проверить указнные в экземплярах данных значения с кодами по этому справочнику, что обозначено соответствующими warning'ами

2) для справочника прописалась ошибочная версия "0.1.0", "publisher" (какой-то example) и статус (draft - статус не может быть черновик, это полноценный справочник из ФНСИ):
![image](https://github.com/fhir-ru/core/assets/92928137/12b03e49-d79b-4c4d-96a4-f05949485ec7)

3) ссылка из справочника (codeSystem) в NamingSystem не работает - отправляет на пустую страницу, хотя страница для NamingSystem есть.

**Reply by Vrodvi** [reply]
4. В namingsystem нет "кликабельной" ссылки на ФНСИ - ни в описании, ни в пречне идентификаторов, см.:
![image](https://github.com/fhir-ru/core/assets/92928137/5fdc3033-f166-40be-8df5-3b75ff3f3fe2)

**Reply by eugenekogan-spb** [reply]
1. Исправлено на 
* ^content = #fragment

Теперь Expansion появилось
<img width="1138" alt="image" src="https://github.com/fhir-ru/core/assets/62525329/6f353dd5-5d54-4e07-8488-0f63cbad5208">



2, 3, 4 относятся к сборке сайта.

**Reply by Vrodvi** [reply]
`#fragment` - не подходит по смыслу из описания в спецификации - см. https://hl7.org/fhir/valueset-codesystem-content-mode.html
![image](https://github.com/fhir-ru/core/assets/92928137/59e116a5-6b15-4bb4-8101-cf32cce29870)
Это значение, для фрагментов справочников, которые авторы справочника выделяют, ведут, а главное явно обозначают и объясняют причину его выделения.
В нашем случае ФНСИ никаких фрагментов справочника не выделяет и справочник существует в цельном виде, так что это значение вводит в заблуждение и противоречит спецификации.

Правильным вариантом будет включение всех кодов из справочника и установка значения #complete.

**Reply by Vrodvi** [reply]
4. не относится к сборке сайта. Там в definition и правда нет ссылки на страницу справочника в ФНСИ (к примеру, на https://nsi.rosminzdrav.ru/dictionaries/1.2.643.5.1.13.13.99.2.48 ), а просто в угловых скобочках указано название справочника - `<Документы, удостоверяющие личность>`
    - в этом месте нужно сделать ссылку
